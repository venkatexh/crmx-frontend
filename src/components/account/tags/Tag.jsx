const Tag = ({ name, length, createdAt, updatedAt }) => {
  return (
    <div>
      <div>
        <div>{name}</div>
        <div>{length} contacts</div>
      </div>
      <div>
        <div>Created at {createdAt}</div>
        <div>Last updated at {updatedAt}</div>
      </div>
      <div>
        <button>View Tag</button>
      </div>
    </div>
  );
};

export default Tag;
