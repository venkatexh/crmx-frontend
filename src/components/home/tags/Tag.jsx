import "../../../sass/components/home/tags/tag.component.scss";
import { memo } from "react";

const Tag = ({ name, length, createdAt, updatedAt }) => {
  return (
    <div className={"tag"}>
      <div className={"left"}>
        <div className={"name"}>{name}</div>
        <div className={"length"}>{length} contacts</div>
      </div>
      <div className={"middle"}>
        <div className={"crated"}>Created at {createdAt}</div>
        <div className={"updated"}>Last updated at {updatedAt}</div>
      </div>
      <div className={"right"}>
        <button className={"button"}>View Tag</button>
      </div>
    </div>
  );
};

export default memo(Tag);
