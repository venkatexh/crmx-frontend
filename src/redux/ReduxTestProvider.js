import { Provider } from "react-redux";

const ReduxTestProvider = ({ children, reduxStore }) => (
  <Provider store={reduxStore}>{children}</Provider>
);

export default ReduxTestProvider;
