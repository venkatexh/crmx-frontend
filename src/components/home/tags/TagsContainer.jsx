import { useDispatch, useSelector } from "react-redux";
import { useEffect, useState } from "react";
import Tag from "./Tag";
import "../../../sass/components/home/campaigns/container.scss";
import { saveTags } from "../../../redux/actions/account/saveTags";

const TagsContainer = () => {
  const dispatch = useDispatch();
  const [loading, setLoading] = useState(false);
  const state = useSelector(({ tags, loggedUser }) => ({
    tags,
    loggedUser,
  }));

  useEffect(() => {
    setLoading(true);
    setTimeout(() => {
      dispatch(
        saveTags(state.loggedUser.id, state.loggedUser.organization._id)
      );
      setLoading(false);
    }, 500);
  }, []);

  const componentToRender = () => {
    if (loading) {
      return (
        <img
          src={"/loaders/comp_loader.gif"}
          alt={"loader"}
          className={"loader"}
        />
      );
    } else {
      return (
        <div>
          {state.tags.length === 0 ? (
            <div>You don't have any tags yet</div>
          ) : (
            <div>
              {state.tags.map(
                ({ _id, name, contacts, createdAt, updatedAt }) => {
                  return (
                    <Tag
                      key={_id}
                      name={name}
                      length={contacts.length}
                      createdAt={createdAt}
                      updatedAt={updatedAt}
                    />
                  );
                }
              )}
            </div>
          )}
        </div>
      );
    }
  };
  return (
    <div className={"campaignsContainer"}>
      <div className={`${loading ? "campaignsLoading" : "campaignsLoaded"}`}>
        {componentToRender()}
      </div>
      <div className={"filtersList"}></div>
    </div>
  );
};

export default TagsContainer;
