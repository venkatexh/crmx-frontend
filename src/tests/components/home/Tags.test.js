import { shallow } from "enzyme";
import React from "react";
import Tag from "../../../components/home/tags/Tag";
import TagsContainer from "../../../components/home/tags/TagsContainer";
import TagsHeader from "../../../components/home/tags/TagsHeader";
import ReduxTestProvider from "../../../redux/ReduxTestProvider";

it("should be tag component snapshot", () => {
  expect(
    shallow(
      <ReduxTestProvider>
        <Tag />
      </ReduxTestProvider>
    )
  ).toMatchSnapshot();
});

it("should be tags container component snapshot", () => {
  expect(
    shallow(
      <ReduxTestProvider>
        <TagsContainer />
      </ReduxTestProvider>
    )
  ).toMatchSnapshot();
});

it("should be tags header component snapshot", () => {
  expect(
    shallow(
      <ReduxTestProvider>
        <TagsHeader />
      </ReduxTestProvider>
    )
  ).toMatchSnapshot();
});
