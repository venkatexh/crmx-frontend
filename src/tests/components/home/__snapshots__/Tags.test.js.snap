// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should be tag component snapshot 1`] = `ShallowWrapper {}`;

exports[`should be tags container component snapshot 1`] = `ShallowWrapper {}`;

exports[`should be tags header component snapshot 1`] = `ShallowWrapper {}`;
